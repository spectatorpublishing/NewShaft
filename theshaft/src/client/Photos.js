import React, { Component } from 'react';

export default class Photos extends Component {

}