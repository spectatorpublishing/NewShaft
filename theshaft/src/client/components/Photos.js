import React, { Component } from "react";
import styled from 'styled-components';

let Imagecontainer = styled.div`
  display: flex;
  flex-direction: row;
  height: 40vh;
  width: 100%;
  overflow: hidden;
  align-items: center;
`

let PicsBig = styled.div`
  display: flex;
  flex-direction: column;
  width: 66%;
  height: 100%;
  overflow: hidden;
  align-items: center;
`

let PicsMid = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 33%;
  overflow: hidden;
`

let PicsLow = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 50%;
  overflow: hidden;
`

let Pinky = styled.img`
  height: 50%;
`

let Inky = styled.img`
  height: 100%;
  width: 50%;
`

let Clyde = styled.img`
  height: 100%;
  width: 50%;
`


export default class Photos extends Component {
  constructor(props) {
    super(props);

    this.state = {
      imageOne: this.props.imageOne,
      imageTwo: this.props.imageTwo,
      imageThree: this.props.imageThree,
      imageFour: this.props.imageFour
    };
  }

  render() {
    return (
      <Imagecontainer>
        <PicsBig>
          <img src={this.state.imageOne} />
        </PicsBig>
        <PicsMid>
          <Pinky src={this.state.imageTwo} />
          <PicsLow>
            <Inky src={this.state.imageThree} />
            <Clyde src={this.state.imageFour} />
          </PicsLow>
        </PicsMid> 
      </Imagecontainer>
    );
  }
}
