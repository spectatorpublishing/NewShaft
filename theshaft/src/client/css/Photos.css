div.picsHigh {
	margin: 3em;
	display: flex;
	flex-direction: row;
	height: 500px;
	width: 1000px;
	overflow: hidden;
}
div.picsMid {
	display: flex;
	flex-direction: column;
	height: 100%;
	width: 33%;
	overflow: hidden;
}
div.picsLow {
	display: flex;
	flex-direction: row;
	width: 100%;
	height: 50%;
	overflow: hidden;
}

img {
	padding: 5px;
}

img.blinky {
	height: 100%;
	width: 66%;
}

img.pinky {
	height: 50%;
}

img.inky {
	height: 100%;
	width: 50%;
}

img.clyde {
	height: 100%;
	width: 50%;
}
