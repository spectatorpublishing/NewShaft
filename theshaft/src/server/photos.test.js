import React from 'react';
import Photos from '../client/components/Photos.js';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Photos imageOne="https://memegenerator.net/img/images/17438601/dat-sad-fat-cat.jpg" imageTwo="https://i.imgflip.com/26a82h.jpg" imageThree="https://i.imgflip.com/1eg7jb.jpg" imageFour="https://i.imgflip.com/1yt82g.jpg"/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});