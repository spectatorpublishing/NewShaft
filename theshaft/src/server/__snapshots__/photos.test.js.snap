// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="picsHigh"
>
  <img
    class="blinky"
    src="https://memegenerator.net/img/images/17438601/dat-sad-fat-cat.jpg"
  />
  <div
    class="picsMid"
  >
    <img
      class="pinky"
      src="https://i.imgflip.com/26a82h.jpg"
    />
    <div
      class="picsLow"
    >
      <img
        class="inky"
        src="https://i.imgflip.com/1eg7jb.jpg"
      />
      <img
        class="clyde"
        src="https://i.imgflip.com/1yt82g.jpg"
      />
    </div>
  </div>
</div>
`;
